cmd/fs-v1-helpers.go

```diff
@@ -386,6 +386,26 @@ func fsFAllocate(fd int, offset int64, len int64) (err error) {
 	return nil
 }
 
+// Renames source path to destination path, fails if the destination path
+// parents are not already created.
+func fsSimpleRenameFile(ctx context.Context, sourcePath, destPath string) error {
+	if err := checkPathLength(sourcePath); err != nil {
+		logger.LogIf(ctx, err)
+		return err
+	}
+	if err := checkPathLength(destPath); err != nil {
+		logger.LogIf(ctx, err)
+		return err
+	}
+
+	if err := os.Rename(sourcePath, destPath); err != nil {
```

Comment:
@harshavardhana Not sure if we need `retry once in case of error` in renames here. as we do currently in `reliableRename()`. Otherwise LGTM.

Reply:
That is historical, because of NFS backends we had to repeat certain operations, especially the ones which are lock free. Requires parents and child directory manipulations.

In this situation we don't need it, as the failure is expected.